pyiceberg/io/pyarrow.py

```diff
@@ -572,51 +572,54 @@ def _convert_scalar(value: Any, iceberg_type: IcebergType) -> pa.scalar:
 
 
 class _ConvertToArrowExpression(BoundBooleanExpressionVisitor[pc.Expression]):
+    def _flat_name_to_list(self, name: str) -> List[str]:
+        return name.split(".")
```

Comment:
This is a risky route. I'd rather migrate to a `Tuple[str]` situation internally so we can actually support fields with `.`'s